packages/@aws-cdk/aws-sagemaker/lib/notebook-instance.ts

```diff
@@ -0,0 +1,237 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+import iam = require('@aws-cdk/aws-iam');
+import kms = require('@aws-cdk/aws-kms');
+import { Construct, Fn, Lazy, Resource } from '@aws-cdk/core';
+import { CfnNotebookInstance, CfnNotebookInstanceLifecycleConfig } from './sagemaker.generated';
+
+/**
+ * Construction properties for a SageMaker Notebook instance
+ *
+ * @experimental
+ */
+export interface NotebookInstanceProps {
+
+    /**
+     * Enable the notebook to have direct internet access.
+     *
+     * @default true
+     */
+    readonly enableDirectInternetAccess?: boolean;
+
+    /**
+     * Instance type of the notebook.
+     *
+     * @default m1.t2.medium
+     */
+    readonly instanceType?: ec2.InstanceType;
+
+    /**
+     * Encryption key for the EBS volume attached to the notebook instance.
+     *
+     * @default none
+     */
+    readonly kmsKeyId?: kms.IKey;
+
+    /**
+     * Name of the notebook instance.
+     *
+     * @default none
```

Comment:
This is the physical name, I believe? So the docs should reflect that, and the default should be `- generated by CloudFormation at deploy time`.